### PR TITLE
installation instructions/crates.io badge

### PR DESCRIPTION
**Pull request by seg6**
Added installation instructions and a crates.io badge/link to the README. Also, @markus-wa can you add me as a collaborator on crates.io (name: brkp) so that I can also push changes to there without bothering you first?

Adresses #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,52 +14,48 @@ Another reason to choose mmproxy-rs may be if you want to avoid interference fro
...
+
+### Installing with Cargo
+
+```sh
+# install from the main branch
+$ cargo install --git https://github.com/saiko-tech/mmproxy-rs
+# install directly from crates.io
+$ cargo install mmproxy
 $ mmproxy -h
-Usage: mmproxy [-h] [options]
```
Comment (markus-wa) [inline README.md:23]: can we add this back? I think it's quite useful

File: README.md [hunk 2]
```diff
@@ -13,52 +14,48 @@ Another reason to choose mmproxy-rs may be if you want to avoid interference fro
 ## Features
 
 - [x] TCP - Accepts PROXY Protocol enabled requests from [Nginx](https://docs.nginx.com/nginx/admin-guide/load-balancer/using-proxy-protocol/#proxy-protocol-for-a-tcp-connection-to-an-upstream), [HAProxy](https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt)
-- [x] UDP - Accepts PROXY Protocol enabled requests from [udppp](https://github.com/b23r0/udppp)
+- [x] UDP - Accepts PROXY Protocol enabled requests from [udppp](https://github.com/b23r0/udppp), [Cloudflare Spectrum](https://www.cloudflare.com/products/cloudflare-spectrum/)
 - [x] No Garbage Collection pauses
 
-## Usage
+## Installation
```
Comment (markus-wa) [inline README.md]: Should be called requirements

File: README.md [hunk 3]
```diff
@@ -13,52 +14,48 @@ Another reason to choose mmproxy-rs may be if you want to avoid interference fro
...
 
+Install Rust with [rustup](https://rustup.rs/) if you haven't already.
+
+```sh
+$ curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+$ cargo --version
 ```
+
+### Installing with Cargo
```
Comment (markus-wa) [inline README.md]: should just be `## Installation`

bring back `## Usage`

File: README.md [hunk 4]
```diff
@@ -13,52 +14,48 @@ Another reason to choose mmproxy-rs may be if you want to avoid interference fro
...
+
+```sh
+# install from the main branch
+$ cargo install --git https://github.com/saiko-tech/mmproxy-rs
```
Comment (markus-wa) [inline README.md]: should be `    cargo install --git https://github.com/saiko-tech/mmproxy-rs`

e.g.

From crates.io:

    cargo install mmproxy

From git:

    cargo install --git https://github.com/saiko-tech/mmproxy-rs

**Comment by markus-wa** [comment]
I've added you on crates.io :slightly_smiling_face:

File: README.md (hunk 1 above)
Comment (seg6) [reply README.md:23]: Sorry, having a busy week. Will get back to this PR once I have some time, thanks for the review!

**Comment by seg6** [comment]
Should be good to merge now.

**Comment by markus-wa** [comment]
thanks - I've moved the install instructions up a bit

**Pull request merged**